transports/upgrades/sqlserver-4to5.md

```diff
@@ -0,0 +1,118 @@
+
+Upgrade a single endpoint at a time. As each endpoint is upgraded to version 5, configure it to operate in backwards compatibility mode to enable it to participate in publish subscribe operations with endpoints that are running on version 4.x or below. An endpoint operating in backwards compatibility mode will check the configured persistence and the native subscriptions table for subscribers when publishing a message. The transport will only send a single copy of the message to each subscriber.
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+
+var pubSubCompat = transport.EnableMessageDrivenPubSubCompatibilityMode();
+```
+
+Configuration APIs to register publishers for event types has been moved to the compatibility mode settings object.
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+var routing = transport.Routing();
+routing.RegisterPublisher(eventType, publisherEndpoint);
+routing.RegisterPublisher(assembly, publisherEndpoint);
+routing.RegisterPublisher(assembly, namespace, publisherEndpoint);
+routing.SubscriptionAuthorizer(authorizationFunction);
+```
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+
+var pubSubCompat = transport.EnableMessageDrivenPubSubCompatibilityMode();
+pubSubCompat.RegisterPublisher(eventType, publisherEndpoint);
+pubSubCompat.RegisterPublisher(assembly, publisherEndpoint);
+pubSubCompat.RegisterPublisher(assembly, namespace, publisherEndpoint);
+pubSubCompat.SubscriptionAuthorizer(authorizationFunction);
+```
+
+NOTE: The `routing.DisablePublishing()` API has been deprecated and should be removed. This API was created to allow an endpoint to run without a configured subscription persistence. On version 5 and above, a subscription persistence is not required unless the endpoint runs in backwards compatibility mode.
+
+Once all endpoint have been upgraded to version 5 and deployed, compatibility mode can be disabled on endpoints, one at a time.
+
+
+### Configure subscription caching
+
+Subscription information can be cached for a given period of time so that it does not have to be loaded every single time an event is being published. The longer the cache period is, the higher the chance that new subscribers miss some events.
+
+Because of that there is no good default value for the subscription caching period. It has to be specified by the user. In systems where subscriptions are static the caching period can be relatively long. To configure it, use following API:
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+var subscriptions = transport.PubSub();
```

Comment:
@SzymonPobiega Should we rename this API to `.SubscriptionSettings()` to match the naming used in SQL Persistence? This would allow us to have `transport.SubscriptionSettings().CacheFor(x)`.

Reply:
I like `SubscriptionSettings` 👍 . I'll change it and try to push SQLT 5